strings: `join`, `to_c_char` (#936)
fortran-lang/stdlib: doc/specs/stdlib_strings.md

{!example/strings/example_zfill.f90!}
```

<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
### `join`

#### Description

Joins an array of strings into a single string. This function concatenates the strings from the input array, 
inserting a separator between each string (default: space). A user-defined separator may be provided, The resulting string is returned.

#### Syntax

> `cmd = ` [[stdlib_strings(module):join(interface)]] ` (strings, separator)`

Reviewer: ```suggestion
`joined = ` [[stdlib_strings(module):join(interface)]] ` (strings, separator)`
```

#### Status

Experimental

#### Class

Pure function

#### Argument

- `strings`: Array of strings (either `type(string_type)` or `character(len=*)`).
  This argument is `intent(in)`. It is an array of strings that will be concatenated together.

> - `separator`: Character scalar (optional).

Author: ```suggestion
- `separator`: `character(len=*)` scalar (optional).
```

```fortran
{!example/strings/example_to_string.f90!}
```

<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->

> ### `to_c_string`

Reviewer 2: As the output is a C character, the  name `to_c_char` might be more appropriate.

Author: [23e535b](https://github.com/fortran-lang/stdlib/pull/936/commits/23e535bc8abea89d2b193961af5030d8a1ec7ddf)

#### Description

Convert a Fortran `character` string or a `type(string_type)` variable to a C character array.  
This function converts a Fortran string into a C-style array of characters, ensuring proper null-termination for use in C functions or libraries.  

#### Syntax

`cstr = ` [[stdlib_strings(module):to_c_string(function)]] ` (value)`

#### Status

Experimental

#### Class

Pure function.

#### Argument

- `value`: Shall be a `character(len=*)` string or a `type(string_type)` variable. It is an `intent(in)` argument.  

>   The Fortran string that will be converted to a C character array.

Reviewer 2: ```suggestion
  This Fortran variable will be converted to a C character array.
```
